packages/babel-plugin-transform-function-sent/src/index.js

```diff
@@ -0,0 +1,47 @@
+import syntaxFunctionSent from "babel-plugin-syntax-function-sent";
+import wrapFunction from "babel-helper-wrap-function";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxFunctionSent,
+
+    visitor: {
+      MetaProperty(path) {
+        if (!t.isIdentifier(path.node.meta, { name: "function" })) return;
+        if (!t.isIdentifier(path.node.property, { name: "sent" })) return;
+
+        const fnPath = path.getFunctionParent();
+
+        if (!fnPath.node.generator) {
+          throw new Error("Parent generator function not found");
+        }
+
+        const sentId =
+          fnPath.getData("sentId") ||
+          fnPath.setData(
+            "sentId",
+            path.scope.generateUidIdentifier("function.sent"),
+          );
+
+        path.replaceWith(sentId);
+      },
+
+      Function: {
```

Comment:
This will hit a lot of unneeded functions. We could just wrap the first time we find a `function.sent`.